### ISSUE TITLE
Slow Training

### ISSUE THREAD

**Issue by Bey3fendi** [pierluigiferrari/ssd_keras#396]
Hi @pierluigiferrari and everyone,

here is the some parameters and informations about my environment which I train SSD300 from scratch with this code:

- model: SSD300
- batch size: 32
- epoch: 120
- step-per-epoch: 1000
- Env: Google Colab env
- GPU: Google Colab T4 16GB GPU
- GPU RAM consumption: approx. 9GB
- System RAM consumption: approx 18 GB
- TensorFlow version: 1.15
- keras version: 2.2.5
- cudnn: _should be_ 10.1

with that environment I train the model with PASCAL VOC 2007+2012(approx 16.5k image) and validate it with PASCAL VOC 2007 test datas.
My total training approx. _**take 38 hour 10 minutes**_ and as you can see from the log below

_**Epoch 00001: LearningRateScheduler setting learning rate to 0.001.
2025-04-09 00:13:02.444583: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2025-04-09 00:13:04.830010: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
1000/1000 [==============================] - 1177s 1s/step - loss: 10.9834 - val_loss: 9.2184

Epoch 00001: val_loss improved from inf to 9.21837, saving model to /content/drive/MyDrive/VOC/saved_checkpoints/ssd300_pascal_07+12_epoch-01_loss-10.9826_val_loss-9.2184.h5
Epoch 2/120

Epoch 00002: LearningRateScheduler setting learning rate to 0.001.
1000/1000 [==============================] - 1144s 1s/step - loss: 9.0110 - val_loss: 8.2928

Epoch 00002: val_loss improved from 9.21837 to 8.29280, saving model to /content/drive/MyDrive/VOC/saved_checkpoints/ssd300_pascal_07+12_epoch-02_loss-9.0113_val_loss-8.2928.h5
Epoch 3/120

Epoch 00003: LearningRateScheduler setting learning rate to 0.001.
1000/1000 [==============================] - 1145s 1s/step - loss: 8.3930 - val_loss: 7.8798**_

every epoch approx take 1144 seconds. I observe some other articles and works and 38 hours seems like very long time for SSD training. is anyone or @pierluigiferrari have any idea why it is slow that much? I can provide extra informations if it is needed.

Thanks and best regards.

Lastly can you also write what is the approx. time your system consume while training SSD300. I try to find information about that from the repo but I assume there is not any info about that. @pierluigiferrari 

[epoch120_finishSSD.txt](https://github.com/user-attachments/files/22171681/epoch120_finishSSD.txt)
[ramandgpuramcosnumption.txt](https://github.com/user-attachments/files/22171682/ramandgpuramcosnumption.txt)
